base_layer/core/src/chain_storage/blockchain_database.rs

```diff
@@ -1758,6 +1761,7 @@ fn rewind_to_height<T: BlockchainBackend>(
         chain_header.accumulated_data().total_accumulated_difficulty
     );
 
+    let mut txn = DbTransaction::new();
```

Comment:
Hmm just thought of something, we should update the best block metadata in the same transaction as we delete a block, otherwise it can get out of sync

Reply:
luckily we already load the header in that loop

Reply:
yeah, I think thats correct - we are saying to other nodes "I have no blocks to offer", and the chain metadata should be set to the genesis height (0), hash and accumulated data